src/Core/Addon/Theme/ThemeManager.php

```diff
@@ -23,10 +23,54 @@
  * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
  * International Registered Trademark & Property of PrestaShop SA
  */
-namespace PrestaShop\PrestaShop\Core\Addon;
+namespace PrestaShop\PrestaShop\Core\Addon\Theme;
+
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Addon\Theme\ThemeChecker;
+use PrestaShop\PrestaShop\Core\Addon\AddonManagerInterface;
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilter;
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilterType;
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilterStatus;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use \Tools;
 
 class ThemeManager implements AddonManagerInterface
 {
+    private $shop;
+    private $employee;
+    private $theme_checker;
+    private $configurator;
+    private $fs;
+    private $finder;
+    private $themes;
+
+    public function __construct(
+        \Shop $shop,
+        ConfigurationInterface $configurator,
+        ThemeChecker $theme_checker,
+        \Employee $employee = null,
```

Comment:
This should be required as the class cannot guess which employee it should use :)